BLSQ/iaso
# IA-2166 Org unit types: refactor

From the dashboard you can edit org unit types and set a circular hierarchy like area / city / area

Could not resist to refactor org unit types domain and remove redux stuf 🥳 


Related JIRA tickets : IA-2166

## Self proofreading checklist


## Changes

- add validation on sub unit types in the api
- added order  on org unit type api
- refactor org unit types domain
- removed projects and org unit types reducer + actions + tests
- replace state with react-query

## How to test

- visit org unit type page and make sure you cannot save a circular hierarchy (Zone => village => zone)
- visit data source and forms  details page to see if everything is working properly


## Print screen / video


https://github.com/BLSQ/iaso/assets/12494624/32bf7129-3655-4b53-b105-4974ae1235bf



## Notes

Columns whit headerInfo prop is not sortable see [here](https://github.com/BLSQ/bluesquare-components/pull/120)
